src/Shared/Buffers.MemoryPool/DiagnosticMemoryPool.cs

```diff
@@ -131,6 +131,8 @@ protected override void Dispose(bool disposing)
                 {
                     SetAllBlocksReturned();
                 }
+
+                _pool.Dispose();
```

Comment:
Was the DiagnosticMemoryPool purposefully not disposing the underlying pool for debugging purposes? This slowly leaks otherwise